ConfigSpace/util.pyx

```diff
@@ -185,11 +185,15 @@ def get_one_exchange_neighbourhood(
                 else:
                     if iteration > 0:
                         break
-                    if hp_name not in finite_neighbors_stack:
+                    neighbors = finite_neighbors_stack.get(hp_name, [])
+                    if len(neighbors) == 0:
                         if isinstance(hp, UniformIntegerHyperparameter):
+                            _n_neighbors = min(n_neighbors_per_hp[hp_name], MAX_NEIGHBORHOOD)
                             neighbors = hp.get_neighbors(
-                                value, random,
-                                number=n_neighbors_per_hp[hp_name], std=stdev,
+                                value,
+                                random,
+                                number=_n_neighbors,
+                                std=stdev

```

Comment:
There's a line further below this block `neighbors.pop()`. After fixing an issue with `UniformIntegerHyperparamter::get_num_Neighbors`, this cause the benchmark to fail. There was some parameter with only 3 possible neighbors which must have been sampled a few times, causing it to run out of neighbors during this procedure.

Now it will just generate a new set of neighbors if it has run out from all of the `neighbors.pop`